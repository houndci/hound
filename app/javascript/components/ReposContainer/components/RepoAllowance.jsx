import React from 'react';

export default class RepoAllowance extends React.Component {
  render() {
    const { subscribedRepoCount, tierAllowance } = this.props;

    return(
      <div className="allowance">
        { "Private Repos " }
        <strong>
          <span data-role="subscribed-repo-count">{subscribedRepoCount}</span>
          { " / " }
          <span data-role="tier-allowance">{tierAllowance}</span>
        </strong>
      </div>
    );
  }
}
