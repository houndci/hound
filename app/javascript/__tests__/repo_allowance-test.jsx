import RepoAllowance from '../components/ReposContainer/components/RepoAllowance';

it('renders the allowance header', () => {
  const wrapper = shallow(
    <RepoAllowance
      subscribedRepoCount={4}
      tierAllowance={10}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
