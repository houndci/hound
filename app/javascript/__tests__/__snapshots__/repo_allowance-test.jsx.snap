exports[`test renders the allowance header 1`] = `
<div
  className="allowance">
  Private Repos 
  <strong>
    <span
      data-role="subscribed-repo-count">
      4
    </span>
     / 
    <span
      data-role="tier-allowance">
      10
    </span>
  </strong>
</div>
`;
